Source/VMware.vSphereDSC/DSCResources/VMHostPermission.ps1

```diff
@@ -0,0 +1,530 @@
+            $entity = Get-Datastore -Server $this.Connection -Name $this.EntityName -ErrorAction SilentlyContinue -Verbose:$false
+        }
+        elseif ($this.EntityType -eq [EntityType]::VM) {
+            <#
+            If the Entity is a VM, the Entity location is either a Resource Pool or a vApp where the VM is placed. If the VMHost is managed by a vCenter,
+            there is a special case where two VMs could be created with the same name on the same VMHost but on different vCenter folders. And this way the
+            VM could not be uniquely identified on the VMHost.
+            #>
+            $entity = Get-VM -Server $this.Connection -Name $this.EntityName -Location $entityLocation -ErrorAction SilentlyContinue -Verbose:$false
+
+            # Only throw an exception if Ensure is 'Present', otherwise ignore that multiple VMs were found.
+            if ($entity.Length -gt 1 -and $this.Ensure -eq [Ensure]::Present) {
+                throw ($this.CouldNotIdentifyVMMessage -f $this.EntityName, $this.Connection.Name, $entity.Length)
+            }
+        }
+        else {
+            <#
+            If the Entity is a Resource Pool or vApp, the Entity location is either a Resource Pool or a vApp where the Entity is placed. For a specific Resource Pool
+            or vApp, the name does not uniquely identify the Entity because there can be other Resource Pools or vApps below in the hierarchy with the same name placed in the
+            Entity location. So additional filtering is needed to verify that the Entity is directly placed in the specified Entity location.
+            #>
+            $entity = Get-ResourcePool -Server $this.Connection -Name $this.EntityName -Location $entityLocation -ErrorAction SilentlyContinue -Verbose:$false |
+                      Where-Object -FilterScript { $_.ParentId -eq $entityLocation.Id }
+        }
+
+        $exceptionMessage = $this.CouldNotFindEntityMessage -f $this.EntityName, $this.EntityType.ToString(), $this.Connection.Name
+        $this.EnsureCorrectBehaviourIfTheEntityIsNotFound($entity, $exceptionMessage)
+
+        return $entity
+    }
+
+    <#
+    .DESCRIPTION
+
+    Retrieves the Principal with the specified name from the VMHost if it exists.
+    If the name contains the '\' sign, it means that the Principal is part of a Domain, so the search for the Principal should be done by filtering by Domain.
+    If Ensure is 'Present' and the Principal is not found, an exception is thrown. If Ensure is 'Absent' and the Principal is not found, $null is returned.
+    #>
+    [PSObject] GetVMHostPrincipal() {
+        $getVIAccountParams = @{
+            Server = $this.Connection
+            Verbose = $false
+        }
+
+        # If the Principal is a Domain User, we should extact the Domain and User names from the Principal name.
+        if ($this.PrincipalName -Match '\\') {
```

Comment:
Check domain\user is always the case. I guess user@domain is also a valid domain user format.

Reply:
Done.